adoc/chapters/architecture.adoc

```diff
@@ -3,36 +3,25 @@
 [[architecture]]
 = SYCL architecture
 
-This chapter describes the structure of a SYCL application, and how the SYCL
-generic programming model lays out on top of a number of <<backend>>s.
+This chapter describes the structure of a SYCL application and its programming
+model.
 
 
 == Overview
 
 SYCL is an open industry standard for programming a heterogeneous system.
 The design of SYCL allows standard {cpp} source code to be written such that it
-can run on either an heterogeneous device or on the <<host>>.
-
-The terminology used for SYCL inherits historically from OpenCL with some
-SYCL-specific additions.
-However SYCL is a generic {cpp} programming model that can be laid out on top of
-other heterogeneous APIs apart from OpenCL.
-SYCL implementations can provide <<backend>>s for various heterogeneous APIs,
-implementing the SYCL general specification on top of them.
-We refer to this heterogeneous API as the <<backend-api>>.
-The SYCL general specification defines the behavior that all SYCL
-implementations must expose to SYCL users for a SYCL application to behave as
-expected.
-
-A function object that can execute on a <<device>> exposed by a <<backend-api>>
-is called a <<sycl-kernel-function>>.
-
-To ensure maximum interoperability with different <<backend-api>>s, software
-developers can access the <<backend-api>> alongside the SYCL general API
-whenever they include the <<backend>> interoperability headers.
-However, interoperability is a <<backend>>-specific feature.
-An application that uses interoperability does not conform to the SYCL general
-application model, since it is not portable across backends.
+can run on either an heterogeneous device such as a GPU or on a general purpose
+CPU.
+
+Some terminology from SYCL derives from OpenCL.
+However, OpenCL is not a prerequisite for implementing SYCL.
+A SYCL implementation may layer on top of a heterogeneous programming API such
+as OpenCL, but this is not necessary.
+It is also possible to implement SYCL directly on hardware.
+
+A function object that can execute on a <<device>> is called a
+<<sycl-kernel-function>>.
```

Comment:
I felt like we needed to provide some example of a "heterogeneous programming API", otherwise no one would know what we mean by that term.  I used OpenCL as the example because that's what the previous text used.  I don't feel that strongly about also adding a reference to CUDA.  We can if we want, but I don't think it adds much extra clarity.  I do *not* think we should list every possible heterogeneous programming API here, though.

Reply:
I agree; it's just we said explicitly that we are not bound to OpenCL and always use OpenCL as an example of a backend.  Just seem a little odd to me, but not important